eve/service/svc_note.py

```diff
@@ -0,0 +1,172 @@
+import json
+from rich.console import Console
+from rich.table import Table
+from rich import box
+import os.path
+from datetime import date
+import pathlib
+
+
+class Note:
+    def __init__(self):
+        # path to JSON-file where to store note-data
+        self.path = str(pathlib.Path.home()) + '/test.json'  # rename for production and make it a hidden file
+
+        # check if JSON-file already exists and if not create it with sample entries
+        if os.path.isfile(self.path):
+            pass
+        else:
+            self.touch = '''[
+                {
+                    "UID": 1,
+                    "CreationDate": "2021-04-24",
+                    "AlterationDate": null,
+                    "DueDate": "2021-04-30",
+                    "Priority": null,
+                    "NoteTxt": "This is sample entry 1."
+                },
+                {
+                    "UID": 2,
+                    "CreationDate": "2021-04-24",
+                    "AlterationDate": "2021-04-24",
+                    "DueDate": "2021-04-29",
+                    "Priority": 1,
+                    "NoteTxt": "This is sample entry 2."
+                },
+                {
+                    "UID": 3,
+                    "CreationDate": "2021-04-24",
+                    "AlterationDate": "2021-04-24",
+                    "DueDate": "2021-04-28",
+                    "Priority": 2,
+                    "NoteTxt": "This is sample entry 3."
+                }
+            ]'''
+            self.touch = json.loads(self.touch)
+            with open(self.path, 'w', encoding='utf-8') as f:
+                json.dump(self.touch, f, indent=2, ensure_ascii=False)
+            print(f'Created {self.path}')
+        # Initialize instance variables
+        self.data = None
+        self.new_note = None
+
+    def output(self):
+        # read JSON-file
+        with open(self.path, encoding='utf-8') as f:
+            self.data = json.load(f)
```

Comment:
It is never good to create an attribute on an object by using `self.data = 'blah'` outside of `__init__`. It mutates the class and means that you've to call the class in a specific order to know if data is populated or not.

Take a look at [this comment](https://github.com/IVIURRAY/yt-eve/pull/5/files#r622444233) about how to structure this to make sure you always have a `self.notes_file`.

Reply:
That makes sense. Thank you.